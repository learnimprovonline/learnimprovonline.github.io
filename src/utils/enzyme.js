import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })
