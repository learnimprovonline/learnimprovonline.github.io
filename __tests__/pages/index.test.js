import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import IndexPage from '../../src/pages/index.js'
import Link from 'gatsby-link'

const activitiesPageDestination = '/activities/'
const pageTitle = 'Early Access Alpha'

const indexPage = shallow(
    <IndexPage />
);

const activitiesLink = indexPage.find(Link);

test('Index Page contains a page heading with the page title', () => {
    const pageHeading = indexPage.find('h1');
    const pageHeadingText = pageHeading.children().text();

    expect(pageHeadingText).toEqual(pageTitle);
});

test('Index Page contains a link to the Activities Page', () => {
    const activitiesLinkDestination = activitiesLink.prop('to');

    expect(activitiesLinkDestination).toEqual(activitiesPageDestination);
});

test('Index Page contains a "Go to Activities" link', () => {
    const activitiesLinkText = activitiesLink.children().text();

    expect(activitiesLinkText).toEqual('Go to Activities');
});

test('Index Page renders correctly', () => {
    const tree = toJson(indexPage);
    
    expect(tree).toMatchSnapshot();
});