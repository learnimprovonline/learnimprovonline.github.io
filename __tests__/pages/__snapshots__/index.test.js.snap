// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders itself 1`] = `
<div>
  <h1>
    Early Access Alpha
  </h1>
  <p>
    Welcome to the future of learning improv. This early access preview gives you an inside look into the exciting new features coming to the site. Keep in mind everything is in active development and will be changing frequently, but don't let that stop you.
  </p>
  <p>
    Now go do something great.
  </p>
  <GatsbyLink
    to="/activities/"
  >
    Go to Activities
  </GatsbyLink>
</div>
`;
